decimal/decimal.go

```diff
@@ -0,0 +1,158 @@
+package decimal
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+	"strconv"
+	"strings"
+)
+
+// Decimal represents a decimal amount, internally stored as a big.Int.
+type Decimal struct {
+	i     *big.Int
+	scale int
+}
+
+// New returns a new Decimal. scale specifies the units of i as an inverse power
+// of 10. For example, a scale of 2 indicates that i is in units of 10^-2, and
+// a value of i=25 would represent 25*10^-2=0.25. If scale=0, then the decimal
+// is an integer.
+func New(i *big.Int, scale int) Decimal {
+	return Decimal{
+		i:     i,
+		scale: scale,
+	}
+}
+
+// MarshalJSON converts the Decimal to JSON bytes.
+func (d Decimal) MarshalJSON() ([]byte, error) {
+	return []byte(`"` + d.String() + `"`), nil
+}
+
+// UnmarshalJSON reads JSON bytes into a Decimal.
+func (d *Decimal) UnmarshalJSON(b []byte) error {
+	s := strings.Trim(string(b), `"`)
+	if strings.IndexByte(s, 'e') > -1 {
+		return errors.New("luno: unsupported decimal notation")
+	}
+	f, err := strconv.ParseFloat(s, 64)
```

Comment:
Conversion to float is lossy. Suggest instead remove the decimal point and use big.Int.UnmarshalText.